src/main/java/ch/njol/skript/expressions/ExprEntityOwner.java

```diff
@@ -0,0 +1,131 @@
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Item;
+import org.bukkit.entity.Tameable;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+@Name("Entity Owner")
+@Description({
+	"The owner of a tameable entity (i.e. horse or wolf) or a dropped item.",
+	"NOTE: If the owner of a dropped item was an entity and was killed, will return their uuid in string form."
+})
+@Examples({
+	"set owner of target entity to player",
+	"delete owner of target entity",
+	"set {_t} to uuid of tamer of target entity",
+	"",
+	"set the owner of all dropped items to player"
+})
+@Since("2.5, INSERT VERSION (dropped items)")
+public class ExprEntityOwner extends SimplePropertyExpression<Entity, Object> {
+	
+	static {
+		Skript.registerExpression(ExprEntityOwner.class, Object.class, ExpressionType.PROPERTY,
+			"[the] (owner|tamer) of %livingentities%",
+			"[the] [[dropped] item] owner of %itementities%");
+	}
+
+	private boolean useTameable;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		useTameable = matchedPattern == 0;
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	public @Nullable Object convert(Entity entity) {
+		if (entity instanceof Tameable tameable && tameable.isTamed()) {
+			return tameable.getOwner();
+		} else if (entity instanceof Item item) {
+			UUID uuid = item.getOwner();
+			if (uuid == null)
+				return null;
+			Entity checkEntity = Bukkit.getEntity(uuid);
+			if (checkEntity != null)
+				return checkEntity;
+			OfflinePlayer checkPlayer = Bukkit.getOfflinePlayer(uuid);
+			if (checkPlayer != null)
+				return checkPlayer;
+			return uuid.toString();
+		}
+		return null;
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return switch (mode) {
+			case SET, DELETE, RESET -> {
+				if (useTameable)
+					yield CollectionUtils.array(OfflinePlayer.class);
+				yield CollectionUtils.array(OfflinePlayer.class, Entity.class, String.class);
+			}
+			default -> null;
+		};
+	}
+	
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		UUID newId = null;
+		OfflinePlayer newPlayer = null;
+		if (delta != null) {
+			if (delta[0] instanceof OfflinePlayer offlinePlayer) {
+				newPlayer = offlinePlayer;
+				newId = offlinePlayer.getUniqueId();
+			} else if (delta[0] instanceof Entity entity) {
+				newId = entity.getUniqueId();
+			} else if (delta[0] instanceof String string) {
+				try {
+					newId = UUID.fromString(string);
+				} catch (Exception ignored) {}
+			}
```

Comment:
is supporting string uuids standard in other entity-related expressions? if not, this should probably be removed

Reply:
in my opinion it should be kept, in the random event someone has a UUID saved in a variable.

Reply:
I'm also in favor  of keeping it.